### PR TITLE
Ad hoc example (Spring)

### PR DESCRIPTION
**Pull request by tacerus**
Signed-off-by: Georg Pfuetzenreuter <mail@georg-pfuetzenreuter.net>

Hi,

this adds a short example on adding executions for the existing `sampleOneTimeTask()`. It additionally shows how to fetch the execution time of a task.

Best,
Georg

### CODE REVIEW

**Review by kagkarlsson — COMMENTED** [review]
Thanks for the contribution! A few suggestions :)

File: examples/spring-boot-example/src/main/java/com/github/kagkarlsson/examples/boot/AdhocController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		AdhocController.scheduler = scheduler;
+	}
+
+	/*
+	 * Add a one-time task by sending a HTTP POST request to /adhoc/<seconds>
+	 */
+	@PostMapping("/adhoc/{expiry}")
+    public String addTask(@PathVariable Integer expiry) {
+		String instanceid = RequestContextHolder.currentRequestAttributes().getSessionId();
+		scheduler.schedule(TaskConfiguration.sampleOneTimeTask().instance(instanceid), Instant.now().plusSeconds(expiry));
```
[the remark this answers is not in the document]
Comment (tacerus) [reply examples/spring-boot-example/src/main/java/com/github/kagkarlsson/examples/boot/AdhocController.java]: Thanks! I pushed an update.

File: examples/spring-boot-example/src/main/java/com/github/kagkarlsson/examples/boot/config/TaskConfiguration.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ Task<Void> recurringSampleTask(CounterService counter) {
      * Define a one-time task which have to be manually scheduled.
      */
     @Bean
-    Task<Void> sampleOneTimeTask() {
+    public static Task<Void> sampleOneTimeTask() {
```
[the remark this answers is not in the document]
Comment (tacerus) [reply examples/spring-boot-example/src/main/java/com/github/kagkarlsson/examples/boot/config/TaskConfiguration.java:55]: Thank you for the feedback.

Unfortunately I don't seem to understand how Spring should inject it - as even if I make `sampleOneTimeTask()` non-static, the reference to it still is static - I attempted your suggestion to create a `TaskInstance` directly, however I realized I would still need to call `TaskConfiguration.sampleOneTimeTask()` within it to pass it on to the scheduler, causing the same issue with it being a static reference.

**Comment by tacerus** [comment]
No worries, thank you!

**Pull request closed without merging**